test/e2e/parallel/space_test.go

```diff
@@ -480,6 +480,39 @@ func TestSubSpaces(t *testing.T) {
 	})
 }
 
+func TestSubSpaceInheritance(t *testing.T) {
+	// given
+	t.Parallel()
+	// make sure everything is ready before running the actual tests
+	awaitilities := WaitForDeployments(t)
+	hostAwait := awaitilities.Host()
+	memberAwait := awaitilities.Member1()
+
+	t.Run("we create a subSpace, with disable inheritance and do not expect roles and usernames from the parent to be inherited in NSTemplateSet", func(t *testing.T) {
+		// when
+		// we have a parentSpace
+		parentSpace, _, parentSpaceBindings := CreateSpace(t, awaitilities, testspace.WithSpecTargetCluster(memberAwait.ClusterName), testspace.WithTierName("appstudio"))
+		// then
+		// wait until MUR and Space have been provisioned
+		VerifyResourcesProvisionedForSpace(t, awaitilities, parentSpace.Name, UntilSpaceHasStatusTargetCluster(memberAwait.ClusterName))
+		mur, err := hostAwait.WaitForMasterUserRecord(t, parentSpaceBindings.Spec.MasterUserRecord)
+		require.NoError(t, err)
+
+		// when
+		// we also have a subSpace with same tier but with disable inheritance
+		subSpace := CreateSubSpace(t, awaitilities, testspace.WithSpecParentSpace(parentSpace.Name), testspace.WithTierName("appstudio"), testspace.WithSpecTargetCluster(memberAwait.ClusterName), testspace.WithDisableInheritance(true))
+		testsupportsb.CreateSpaceBinding(t, awaitilities.Host(), mur, subSpace, "viewer")
```

Comment:
This technically creates a new SpaceBinding for the sub-space and for the same user that has access to the parent space. In other words, the test will pass no matter if the inheritance is disabled or not.

What you could do instead is to provision a brand new user and share the sub-space with this user
```suggestion
		_, secondMur := NewSignupRequest(awaitilities).
			ManuallyApprove().
			EnsureMUR().
			RequireConditions(ConditionSet(Default(), ApprovedByAdmin())...).
			Execute(t).Resources()
		testsupportsb.CreateSpaceBinding(t, awaitilities.Host(), secondMur, subSpace, "contributor")
```

apart from that, there is no "viewer" role in the appstudio NSTemplateTier - let's use "contributor" for example

Reply:
done